xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/changesinline.vm

```diff
@@ -64,5 +64,9 @@
     #end
   #end
 #end
+#set ($previousOrigdocVersion = $origdoc.previousVersion)
+#set ($previousNewdocVersion = $newdoc.previousVersion)
+#set ($nextOrigdocVersion = $origdoc.documentArchive.getNextVersion($origdoc.RCSVersion))
```

Comment:
Did you check if this is working for extension version ?

Reply:
I'm not really discussion the theory here. I just want to make sure @atallahade tested this use case and those lines don't crash or end up providing the 2.1 version of the current document as next version (which would not make any sense) when dealing with non-history documents since manipulating the archive of this kind of document never really been a use case before.

Reply:
@tmortagne I tried the feature with an extension. It seems that the extension version is not reachable when comparing two normal versions (made on the wiki). For instance we can have this kind of history:
```
1.1 <--> 1.2 <--> 2.1 <--> 2.2
```
An extension version will have this type of name `xar:org.xwiki.contrib:application-help-center/1.2` and will put itself before a version, like this:
```
                                 1.1 <--> 1.2 <--> 2.1 <--> 2.2
xar:org.xwiki.contrib:application-help-center/1.2 --^
```

If you compare `xar:org.xwiki.contrib:application-help-center/1.2` to `2.1`, you will not be able to access previous changes (no left arrows will be displayed). Other arrows work fine.

Reply:
> An extension version will have this type of name xar:org.xwiki.contrib:application-help-center/1.2 and will put itself before a version, like this:

Not really sure what you mean here. The extension version is always displayed first in the history UI and no code produce an ordered list with the extension version in the middle of it (that would not make any sense) unless it's a new thing you added.

> If you compare xar:org.xwiki.contrib:application-help-center/1.2 to 2.1, you will not be able to access previous changes (no left arrows will be displayed). Other arrows work fine.

IMO there should be either no arrow at all (comparing with an extension version disable previous/next feature) or have previous and next only apply to the non extension version (i.e. compare the extension version with previous or next document version).

Reply:
Fixed by #719

Reply:
@tmortagne, @atallahade Did you guys reach a conclusion?

AFAIU, right now, there would still be a changes arrow (left or right), next to the document version. So if you are comparing `xar:org.xwiki.contrib:application-help-center/1.2` to `2.1`, you would have a "nextChange" arrow, on the outer right side. Clicking it, would lead to comparing `2.1` to `2.2`.

I personally don't find that too bad, only that it would be lacking a bit of consistency. If you can go to the next change, why not be able to go to the previous change as well? What would that mean in this context? I would say that, if we can only stay relative to the only document version we have (i.e. `2.1`), previous change (i.e. left outer arrow) should lead to comparing `1.2` to `2.1` (i.e. "ignoring" the extension version completely).

Does this make sense?

Reply:
It's kind of the same problem as when the "From" version is higher than the "To" version (i.e. comparing `2.1` to `1.2`), where should the next change button lead to?
For the case of comparing an extension version, I think there should be just no changes arrow at all.
As for the arrows around the extension version, they will never be display as an extension doesn't have a previous/next version anymore.